File: Python/806.py
width = [7,5,4,7,10,7,9,4,8,9,6,5,4,2,3,10,9,9,3,7,5,2,9,4,8,9]
S = "zlrovckbgjqofmdzqetflraziyvkvcxzahzuuveypqxmjbwrjvmpdxjuhqyktuwjvmbeswxuznumazgxvitdrzxmqzhaaudztgie"
width = [10,10,10,10,10,10,10,10,10,10,10,10,10,10,10,10,10,10,10,10,10,10,10,10,10,10]
S = "abcdefghijklmnopqrstuvwxyz"
print len(S)
line = 1
cnt = 0
for i in range(0, len(S)):
	print ord(S[i]) - 97
	if 100-cnt > width[ord(S[i]) - 97]:
		cnt += width[ord(S[i]) - 97]
	elif 100 - cnt == width[ord(S[i]) - 97]:
		cnt = 0
		line += 1;
	else:
		cnt = width[ord(S[i]) - 97]
		line += 1;
	
print line, cnt

File: Java/806.java
public int[] numberOfLines(int[] widths, String S){
	char[] str = S.toCharArray();
	int cnt = 0;
	int line = 1;
	int i = 0;
	for (i = 0; i < str.length; i++){
		char s = str[i];
		
		if (100 - cnt > widths[(byte)s - 97]){
			cnt = cnt + widths[(byte)s - 97];
		}
		else if (100 -cnt == widths[(byte)s - 97]){
			cnt = 0;
			line ++;
		}
		else{
			cnt = widths[(byte)s - 97];
			line ++;
		}
	}
	int[] results = new int[2];
	
	results[0] = line;
	results[1] = cnt;

	return results;
}
